lib/ansible/modules/cloud/amazon/s3.py

```diff
@@ -326,8 +326,8 @@ def create_bucket(module, s3, bucket, location=None):
 
 def list_keys(module, s3, bucket, prefix, marker, max_keys):
     paginator = s3.get_paginator('list_objects')
-    all_keys = [page['Contents'] for page in paginator.paginate(Bucket=bucket)][0]
-    keys = [{'Key': data['Key']} for data in all_keys]
+    all_keys = [page for page in paginator.paginate(Bucket=bucket)][0].get('Contents', [])
```

Comment:
Won't this just get the contents of the first page? I think you need to flatten this out by extending all `contents` to one list.